pkg/controller/download_request_controller_test.go

```diff
@@ -18,306 +18,237 @@ package controller
-			downloadRequest: newDownloadRequest("", velerov1api.DownloadTargetKindRestoreResults, "a-backup-20170912150214"),
+			expectedRequeue: ctrl.Result{Requeue: true},
+		}),
+		Entry("restore results request with phase '' gets a url", request{
+			downloadRequest: builder.ForDownloadRequest(velerov1api.DefaultNamespace, "a-download-request").Phase("").Target(velerov1api.DownloadTargetKindRestoreResults, "a-backup-20170912150214").Result(),
 			restore:         builder.ForRestore(velerov1api.DefaultNamespace, "a-backup-20170912150214").Phase(velerov1api.RestorePhaseCompleted).Backup("a-backup").Result(),
 			backup:          defaultBackup(),
-			backupLocation:  newBackupLocation("a-location", "a-provider", "a-bucket"),
+			backupLocation:  builder.ForBackupStorageLocation(velerov1api.DefaultNamespace, "a-location").Provider("a-provider").Bucket("a-bucket").Result(),
 			expectGetsURL:   true,
-		},
-		{
-			name:            "restore results request with phase 'New' gets a url",
-			downloadRequest: newDownloadRequest(velerov1api.DownloadRequestPhaseNew, velerov1api.DownloadTargetKindRestoreResults, "a-backup-20170912150214"),
+			expectedRequeue: ctrl.Result{Requeue: true},
+		}),
+		Entry("restore results request with phase 'New' gets a url", request{
+			downloadRequest: builder.ForDownloadRequest(velerov1api.DefaultNamespace, "a-download-request").Phase(velerov1api.DownloadRequestPhaseNew).Target(velerov1api.DownloadTargetKindRestoreResults, "a-backup-20170912150214").Result(),
 			restore:         builder.ForRestore(velerov1api.DefaultNamespace, "a-backup-20170912150214").Phase(velerov1api.RestorePhaseCompleted).Backup("a-backup").Result(),
 			backup:          defaultBackup(),
-			backupLocation:  newBackupLocation("a-location", "a-provider", "a-bucket"),
+			backupLocation:  builder.ForBackupStorageLocation(velerov1api.DefaultNamespace, "a-location").Provider("a-provider").Bucket("a-bucket").Result(),
 			expectGetsURL:   true,
-		},
-		{
-			name:            "request with phase 'Processed' is not deleted if not expired",
-			downloadRequest: newDownloadRequest(velerov1api.DownloadRequestPhaseProcessed, velerov1api.DownloadTargetKindBackupLog, "a-backup-20170912150214"),
+			expectedRequeue: ctrl.Result{Requeue: true},
+		}),
+		Entry("request with phase 'Processed' is not deleted if not expired", request{
+			downloadRequest: builder.ForDownloadRequest(velerov1api.DefaultNamespace, "a-download-request").Phase(velerov1api.DownloadRequestPhaseProcessed).Target(velerov1api.DownloadTargetKindBackupLog, "a-backup-20170912150214").Result(),
 			backup:          defaultBackup(),
-		},
-		{
-			name:            "request with phase 'Processed' is deleted if expired",
-			downloadRequest: newDownloadRequest(velerov1api.DownloadRequestPhaseProcessed, velerov1api.DownloadTargetKindBackupLog, "a-backup-20170912150214"),
+			expectedRequeue: ctrl.Result{Requeue: true},
+		}),
+		Entry("request with phase 'Processed' is deleted if expired", request{
```

Comment:
Further to @nrb's comments, we should add some more test cases here to show that if a request has phase 'New', or is empty, that we should still delete it if it has expired. I was going through some of the test logic and it looks like it only applies expiration logic in the case where the status is Processed.